services/api.py

```diff
@@ -816,20 +816,38 @@ def service_nodes_by_ancestors(service_node_ids):
                 bbox_geometry_filter = munigeo_api.build_bbox_filter(ref, val, 'geometry')
                 queryset = queryset.filter(Q(**bbox_filter) | Q(**bbox_geometry_filter))
 
+        def get_category_ids(categories, category):
+            ids = []
+            for mapping in categories:
+                if mapping.split(':')[0] == category:
+                    ids.append(mapping.split(':')[1])
+            return ids
+
         if 'category' in filters:
             services_and_service_nodes = filters.get('category', None).split(',')
-            service_ids = []
-            servicenode_ids = []
-            for category in services_and_service_nodes:
-                key = category.split(':')[0]
-                value = category.split(':')[1]
-                if key == 'service':
-                    service_ids.append(value)
-                elif key == 'service_node':
-                    servicenode_ids.append(value)
+            service_ids = get_category_ids(services_and_service_nodes, 'service')
+            servicenode_ids = get_category_ids(services_and_service_nodes, 'service_node')
             queryset = queryset.filter(Q(services__in=service_ids)
                                        | Q(service_nodes__in=service_nodes_by_ancestors(servicenode_ids))).distinct()
 
+        if 'period' in filters:
+            period = filters.get('period', None)
```

Comment:
Add a TODO note here: Add validation.
It's best to do after the validation framework has been merged.